src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -4375,6 +4375,73 @@ class MavlinkStreamCameraCapture : public MavlinkStream
 	}
 };
 
+class MavlinkStreamGpsGlobalOrigin : public MavlinkStream
+{
+public:
+	const char *get_name() const override
+	{
+		return MavlinkStreamGpsGlobalOrigin::get_name_static();
+	}
+
+	static constexpr const char *get_name_static()
+	{
+		return "GPS_GLOBAL_ORIGIN";
+	}
+
+	static constexpr uint16_t get_id_static()
+	{
+		return MAVLINK_MSG_ID_GPS_GLOBAL_ORIGIN;
+	}
+
+	uint16_t get_id() override
+	{
+		return get_id_static();
+	}
+
+	static MavlinkStream *new_instance(Mavlink *mavlink)
+	{
+		return new MavlinkStreamGpsGlobalOrigin(mavlink);
+	}
+
+	unsigned get_size() override
+	{
+		return _vehicle_local_position_sub.advertised() ?
+		       (MAVLINK_MSG_ID_GPS_GLOBAL_ORIGIN_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES) : 0;
+	}
+
+private:
+	uORB::Subscription _vehicle_local_position_sub{ORB_ID(vehicle_local_position)};
+
+	/* do not allow top copying this class */
+	MavlinkStreamGpsGlobalOrigin(MavlinkStreamGpsGlobalOrigin &) = delete;
+	MavlinkStreamGpsGlobalOrigin &operator = (const MavlinkStreamGpsGlobalOrigin &) = delete;
+
+protected:
+	explicit MavlinkStreamGpsGlobalOrigin(Mavlink *mavlink) : MavlinkStream(mavlink)
+	{}
+
+	bool send() override
+	{
+		vehicle_local_position_s vehicle_local_position;
+
+		if (_vehicle_local_position_sub.update(&vehicle_local_position)) {
+			mavlink_gps_global_origin_t msg{};
+
+			if (vehicle_local_position.ref_timestamp > 0) {
```

Comment:
```suggestion
			if (vehicle_local_position.ref_timestamp != 0 && vehicle_local_position.xy_global && vehicle_local_position,z_global) {
```

Reply:
I thought about that but does it really change the reference? Probably better to make it clear the reference is not used in this moment. 👍

Reply:
Checking only `xy_global` and ignoring the timestamp would be fine.